crates/wasi-common/src/sys/windows/hostcalls_impl/fs.rs

```diff
@@ -513,9 +543,13 @@ pub(crate) fn path_symlink(old_path: &str, resolved: PathGet) -> Result<()> {
                         symlink_dir(old_path, new_path).map_err(Into::into)
                     }
                     WinError::ERROR_ACCESS_DENIED => {
-                        // does the target exist?
                         if new_path.exists() {
+                            // the target already exists
                             Err(Error::EEXIST)
+                        } else if !old_path.exists() {
+                            // attempt to create a dangling symlink
+                            warn!("Dangling symlink or symlink loops unsupported on Windows");
+                            Err(Error::ENOTSUP)
```

Comment:
This looks like a good choice of errno value here, but I'd like @sunfishcode to double check that though.